### PR TITLE
Add support for dynamic feature modules on Android projects

### PR DESCRIPTION
**Pull request by barriosnahuel**
## Description

Added support for Android modules applying:

```
apply plugin: 'com.android.dynamic-feature'
```

Also added a basic `CONTRIBUTING.md` file to help contributors test their changes. See it [on action](https://github.com/barriosnahuel/static-code-analysis-plugin/blob/feature/android-dynamic-feature-modules-support/CONTRIBUTING.md).

## Current behaviour
We can use SCA either on Android app modules or library modules, but it's not working for dynamic feature modules.

- It doesn't make the build fail running: `./gradlew check`, it simply does nothing because applied plugin is not supported yet.
- It fails when running explicitly:

```
> ./gradlew myfeature:checkstyle                                                                                           

> Configure project :app

FAILURE: Build failed with an exception.

* What went wrong:
Task 'checkstyle' not found in project ':myfeature'.

* Try:
Run gradlew tasks to get a list of available tasks. Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 4s

```

## Expected behaviour
Plugin should setup all tools for any kind of Android project

## Steps to reproduce
Replace any of your Android modules from:
```
apply plugin: 'com.android.library'
```
to
```
apply plugin: 'com.android.dynamic-feature'
```

Then execute `./gradlew tasks` and neither Checkstyle nor PMD or Findbugs are gonna be printed.
Also, Android Lint is not honoring the specified config file at `staticCodeAnalysis.androidLintConfig`.

## Post merge behaviour
We will be able to run Checkstyle and other tools in this kind of Android modules.

Output:
```
> ./gradlew myfeature:checkstyle                                                                                           

> Configure project :app

BUILD SUCCESSFUL in 19s
9 actionable tasks: 2 executed, 7 up-to-date
```
## Tests
Manually on a real project using the following config:

```
buildscript {
    repositories {
        jcenter()
        google()
        maven {
            url 'https://maven.fabric.io/public'
        }
        flatDir dirs: 'libs'
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.6.0'
        classpath 'com.google.gms:google-services:4.3.3'
        classpath 'io.fabric.tools:gradle:1.31.2'

        classpath 'com.google.firebase:perf-plugin:1.3.1'
        classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.3.61'

        classpath 'com.monits:static-code-analysis-plugin:2.7.1-SNAPSHOT'
        classpath 'io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.0.1'
        classpath 'com.getkeepsafe.dexcount:dexcount-gradle-plugin:1.0.2'
    }
}
```

I tried to add tests but I found this TODO: https://github.com/Monits/static-code-analysis-plugin/blob/1f417843330d886f4c9465ff13dddea2f42a6b32/src/integTest/groovy/com/monits/gradle/sca/AndroidLintIntegTest.groovy#L71

### CODE REVIEW

**Comment by barriosnahuel** [comment]
@jsotuyod would you mind to re-run the failed job? `com.monits.gradle.sca.performance.PmdPerfTest > Java analysis performance with gradle 5.6.4` failed but it passes locally

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+### Publish a local version
+1. Check plugin version at [build.gradle@version](/build.gradle). i.e. `2.7.1-SNAPSHOT
+2. Build the plugin
+```
+./gradlew assemble
+```
+
+### Add it to a real project
+1. Add SCA's build dir to your projec's buildscript at your root `build.gradle`.
```
Comment (jsotuyod) [inline CONTRIBUTING.md]:
```suggestion
1. Add SCA's build dir to your project's buildscript at your root `build.gradle`.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+```groovy
+buildscript {
+    repositories {
+        // ...
+        flatDir dirs: '/Users/ltorvalds/Projects/static-code-analysis-plugin/build/libs'``
+        // ...
+    }
+```
+
+2. Specify your new local version when declaring the dependency.
```
Comment (jsotuyod) [inline CONTRIBUTING.md]: what's the point of this? Step 1 should already cover it in a far better way

**Review by jsotuyod — COMMENTED** [review]
Thanks for the PR. This looks really good. I'll take care of integration tests as part of the current development track. Notice however comments to `CONTRIBUTING.md`

File: CONTRIBUTING.md (hunk 2 above)
Comment (barriosnahuel) [reply CONTRIBUTING.md]: You're right! It was late 🌃

**Comment by barriosnahuel** [comment]
@jsotuyod you're welcome! It's ready to merge now. Thanks!

**Pull request merged**